src/core/Akka/Actor/ActorCell.DefaultMessages.cs

```diff
@@ -132,27 +132,38 @@ protected internal virtual void AutoReceiveMessage(Envelope envelope)
 
             switch (message)
             {
+                case ActorSelectionMessage selectionMessage:
+                    ReceiveSelection(selectionMessage);
+                    break;
+                case Identify identify:
+                    HandleIdentity(identify);
+                    break;
                 case Terminated terminated:
                     ReceivedTerminated(terminated);
                     break;
+                case PoisonPill _:
+                    HandlePoisonPill();
+                    break;
                 case AddressTerminated terminated:
                     AddressTerminated(terminated.Address);
                     break;
                 case Kill _:
                     Kill();
                     break;
-                case PoisonPill _:
-                    HandlePoisonPill();
-                    break;
-                case ActorSelectionMessage selectionMessage:
-                    ReceiveSelection(selectionMessage);
-                    break;
-                case Identify identify:
-                    HandleIdentity(identify);
+                case Akka.Actor.IntentionalRestart:
```

Comment:
Added restart handling here - it just throws an `IntentionalActorRestartException`